core/src/main/java/io/questdb/cairo/wal/seq/SequencerMetadata.java

```diff
@@ -238,14 +244,16 @@ private void copyFrom(TableStructure tableStruct, TableToken tableToken, int tab
     private void loadSequencerMetadata(MemoryMR metaMem) {
         columnMetadata.clear();
         columnNameIndexMap.clear();
+        readColumnOrder.clear();
 
         try {
-            final long memSize = checkMemSize(metaMem, SEQ_META_OFFSET_COLUMNS);
+            final long memSize = Math.min(checkMemSize(metaMem, SEQ_META_OFFSET_COLUMNS), metaMem.getLong(SEQ_META_OFFSET_WAL_LENGTH));
```

Comment:
The length value should have `int` size since it's written as `metaMem.putInt(0, (int) metaMem.getAppendOffset());`:
```suggestion
            final long memSize = Math.min(checkMemSize(metaMem, SEQ_META_OFFSET_COLUMNS), metaMem.getInt(SEQ_META_OFFSET_WAL_LENGTH));
```